File: src/weather.svc/Controllers/SampleDataController.fs
namespace weather.svc.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc

type WeatherForecast () =

    member val DateFormatted = "" with get, set

    member val TemperatureC = 0 with get, set

    member val Summary = "" with get, set

    member this.TemperatureF with get () = 32 + int (float this.TemperatureC / 0.5556)

[<Route("api/[controller]")>]
type SampleDataController () =
    inherit Controller()

    static let Summaries = 
        [|"Freezing"; "Bracing"; "Chilly"; "Cool"; "Mild"; "Warm"; "Balmy"; "Hot"; "Sweltering"; "Scorching"|]

    [<HttpGet("[action]")>]
    member this.WeatherForecasts() =
        let rng = new Random()
        seq { for i in 1 .. 15 do yield new WeatherForecast (DateFormatted = DateTime.Now.AddDays(float i).ToString("d"), TemperatureC = rng.Next(-20, 55), Summary = Summaries.[rng.Next(Summaries.Length)]) }

File: src/web/Controllers/SampleDataController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Net.Http;
using System.Net.Http.Headers;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Http;

namespace web.Controllers
{
    [Route("api/[controller]")]
    public class SampleDataController : Controller
    {
		private HttpClient _httpClient = new HttpClient() {BaseAddress = new Uri(Environment.GetEnvironmentVariable("WEATHER_SERVICE_ENDPOINT"))};
		
        [HttpGet("[action]")]
        public async Task WeatherForecasts()
        {
            var queryString = Request.QueryString;
			var response = await _httpClient.GetAsync(queryString.Value);
			var content = await response.Content.ReadAsStringAsync();

			Response.StatusCode = (int)response.StatusCode;
			Response.ContentType = response.Content.Headers.ContentType.ToString();
			Response.ContentLength = response.Content.Headers.ContentLength;

			await Response.WriteAsync(content);
        }
    }
}
